app/api/users_groups_roles.py

```diff
@@ -29,6 +35,69 @@ def query(self, view_kwargs):
     }
 
 
+class UsersGroupsRolesListPost(ResourceList):
+    """
+    Create users groups roles
+    """
+
+    def before_post(self, args, kwargs, data):
+        """
+        before get method to get the resource id for fetching details
+        :param args:
+        :param kwargs:
+        :param data:
+        :return:
+        """
+        require_relationship(['group', 'role'], data)
+        group = Group.query.get(data['group'])
+        role = Role.query.get(data['role'])
+        if group.user != current_user:
+            raise ForbiddenError({'pointer': 'group'}, 'Owner access is required.')
+        if role.name != 'organizer':
+            raise ForbiddenError({'pointer': 'role'}, 'Only organizer role is allowed.')
+
+    def before_create_object(self, data, view_kwargs):
+        """
+        before create object method for UsersGroupsRolesListPost Class
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        role_already_exists = UsersGroupsRoles.query.filter_by(
+            email=data['email'], group_id=data['group'], role_id=data['role']
+        ).count()
+        if role_already_exists:
+            raise ConflictError(
+                {'source': '/data'}, 'Role Invite has already been sent for this email.'
+            )
+        user = User.query.filter_by(email=data['email']).first()
+        if user:
+            data['user_id'] = user.id
+
+    def after_create_object(self, users_groups_role, data, view_kwargs):
+        """
+        after create object method for group_role_invite link
+        :param role_invite:
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        users_groups_role.send_invite()
+
+    view_kwargs = True
+    decorators = (jwt_required,)
+    methods = ['POST']
+    schema = UsersGroupsRolesSchema
+    data_layer = {
+        'session': db.session,
+        'model': UsersGroupsRoles,
+        'methods': {
+            'before_create_object': before_create_object,
+            'after_create_object': after_create_object,
+        },
+    }
+
+
 class UsersGroupsRolesDetail(ResourceDetail):
```

Comment:
Don't allow patch. Only GET and delete

Add permission check that only owner can delete this